### PR TITLE
Stream: add set_read/write_timeout functions

### PR DESCRIPTION
**Pull request by chisa0a**
Adds functions to set the read and write timeout on the inner TcpStream.

### CODE REVIEW

**Comment by chisa0a** [comment]
Ping @hashmap for review.

File: src/sam.rs [hunk 1]
```diff
@@ -256,6 +257,20 @@ impl StreamConnect {
 			local_port: self.local_port,
 		})
 	}
+
+  pub fn set_read_timeout(&self, duration: Option<Duration>) -> Result<(), Error> {
+    match self.sam.conn.set_read_timeout(duration) {
+      Ok(_) => Ok(()),
```
Comment (hashmap) [inline src/sam.rs]: Nitpick - probably we could replace `match` here and below with `map_err` or smth like that, we just need to convert error type here if I got it correctly

Comment (chisa0a) [reply src/sam.rs]: No worries, I can make + test the changes. Thanks for the review.

**Comment by chisa0a** [comment]
@hashmap I made the suggested changes (much cleaner). Everything appears to work.

**Pull request merged**